src/types/onyx/Login.ts

```diff
@@ -1,24 +1,24 @@
 import type * as OnyxCommon from './OnyxCommon';
 
-type Login = {
-    /** Phone/Email associated with user */
-    partnerUserID?: string;
+type Login = OnyxCommon.OnyxValueWithOfflineFeedback<
+    {
+        /** Phone/Email associated with user */
+        partnerUserID?: string;
 
-    /** Value of partner name */
-    partnerName?: string;
+        /** Value of partner name */
+        partnerName?: string;
 
-    /** Date login was validated, used to show info indicator status */
-    validatedDate?: string;
+        /** Date login was validated, used to show info indicator status */
+        validatedDate?: string;
 
-    /** Whether the user validation code was sent */
-    validateCodeSent?: boolean;
+        /** Whether the user validation code was sent */
+        validateCodeSent?: boolean;
 
-    /** Field-specific server side errors keyed by microtime */
-    errorFields?: OnyxCommon.ErrorFields;
-
-    /** Field-specific pending states for offline UI status */
-    pendingFields?: OnyxCommon.PendingFields;
-};
+        /** Field-specific server side errors keyed by microtime */
+        errorFields?: OnyxCommon.ErrorFields;
+    },
+    'defaultLogin' | 'validateLogin' | 'addedLogin' | 'deletedLogin'
```

Comment:
Sorry, I'm confused what the 2nd template argument for `OnyxValueWithOfflineFeedback` is for. Could you help explain? Maybe a README example [here](https://github.com/Expensify/App/blob/fdc5cfb0bc1a8de213b6b760a0f4b11e1cb16ddf/contributingGuides/OFFLINE_UX.md#b---optimistic-with-feedback-pattern) could be helpful too?

Reply:
2nd template argument for `OnyxValueWithOfflineFeedback` are additional keys to be provided as keys for `pendingFields`. So these keys `'defaultLogin' | 'validateLogin' | 'addedLogin' | 'deletedLogin'` are outside of the Login model, but can be in pendingFields keys.

So for example, `validateLogin` can be used in pendingFields keys, but it doesn't exist in the Login model
https://github.com/Expensify/App/blob/ba7f01d8b84968af3d263d0c132539f5f9b70f39/src/libs/actions/User.ts#L379-L381


That's also was explained in [this comment](https://github.com/Expensify/App/pull/34799#issuecomment-1924085052) in _1st variant of implementation_ section.

Maybe it will be better to add a clarifying comment right next to the OnyxValueWithOfflineFeedback type, this way it should be easy to find and helpful.

Reply:
@roryabraham Bump for reviewing

Reply:
thanks for explaining